File: Java/java_ai_generated_codes/src/MediumLevel/chatGPT/PrimorialCalculator.java
package MediumLevel.chatGPT;

import java.util.ArrayList;
import java.util.List;

public class PrimorialCalculator {
    public static void main(String[] args) {
        int n = 3; // Change this value to calculate the Primorial for a different 'n'.
        long primorial = calculatePrimorial(n);
        System.out.println("Primorial(" + n + ") = " + primorial);
    }

    public static boolean isPrime(int num) {
        if (num <= 1) {
            return false;
        }
        if (num <= 3) {
            return true;
        }
        if (num % 2 == 0 || num % 3 == 0) {
            return false;
        }
        for (int i = 5; i * i <= num; i += 6) {
            if (num % i == 0 || num % (i + 2) == 0) {
                return false;
            }
        }
        return true;
    }

    public static long calculatePrimorial(int n) {
        List<Integer> primeNumbers = new ArrayList<>();
        int num = 2; // Start with the first prime number

        while (primeNumbers.size() < n) {
            if (isPrime(num)) {
                primeNumbers.add(num);
            }
            num++;
        }

        long primorial = 1;
        for (int prime : primeNumbers) {
            primorial *= prime;
        }

        return primorial;
    }
}

File: Scala/scala_ai_generated_codes/src/main/scala/MediumLevel/chatGPT/PrimorialCalculator.scala
package MediumLevel.chatGPT

object PrimorialCalculator {
  // Function to check if a number is prime
  def isPrime(num: Int): Boolean = {
    if (num <= 1) false
    else if (num == 2) true
    else if (num % 2 == 0) false
    else {
      var i = 3
      while (i * i <= num) {
        if (num % i == 0) return false
        i += 2
      }
      true
    }
  }

  // Function to calculate the Primorial
  def calculatePrimorial(n: Int): Long = {
    var primorial: Long = 1
    var primeCount = 0
    var num = 2

    while (primeCount < n) {
      if (isPrime(num)) {
        primorial *= num
        primeCount += 1
      }
      num += 1
    }

    primorial
  }

  def main(args: Array[String]): Unit = {
    val n = 3 // You can change n to any positive integer
    val primorial = calculatePrimorial(n)
    println(s"The Primorial of $n is $primorial")
  }
}
